lib/sqlalchemy/ext/automap.py

```diff
@@ -1129,24 +1247,28 @@ def _relationships_for_fks(
                     **o2m_kws,
                 )
                 if rel is not None:
+                    rel = cast(Relationship[Any], rel)
```

Comment:
**mike bayer** (zzzeek) wrote:

we can get around this cast by using `@overload` for generate_relationship

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4271